### PR TITLE
Updating SystemMessage action to use doExecute()

### PR DESCRIPTION
**Pull request by ndeet**
See d.o. issue https://www.drupal.org/node/2572119

### CODE REVIEW

File: src/Plugin/RulesAction/SystemMessage.php [hunk 1]
```diff
@@ -37,14 +37,20 @@
 class SystemMessage extends RulesActionBase {
 
   /**
-   * {@inheritdoc}
+   * Set a system message.
+   *
+   * @param string $message
+   *    Message string that should be set.
```
Comment (klausi) [inline src/Plugin/RulesAction/SystemMessage.php]: Indentation should be 2 spaces, not 3.

**Comment by klausi** [comment]
duplicate

**Comment by ndeet** [comment]
Please reopen as the title was unintentionally the same for #248. This one is the correct SystemMessage pull request.

**Comment by ndeet** [comment]
Fixed spaces.

File: src/Plugin/RulesAction/SystemMessage.php [hunk 2]
```diff
@@ -37,14 +37,20 @@
...
+   *
+   * @param string $message
+   *   Message string that should be set.
+   * @param string $type
+   *   Type of the message.
+   * @param boolean $repeat
+   *   Boolean if the message should be repeated.
    */
-  public function execute() {
+  protected function doExecute($message, $type, $repeat) {
```
Comment (klausi) [inline src/Plugin/RulesAction/SystemMessage.php:49]: $repeat is optional, so should have default value of NULL

File: src/Plugin/RulesAction/SystemMessage.php [hunk 3]
```diff
@@ -37,14 +37,20 @@
...
 
   /**
-   * {@inheritdoc}
+   * Set a system message.
+   *
+   * @param string $message
+   *   Message string that should be set.
+   * @param string $type
+   *   Type of the message.
+   * @param boolean $repeat
```
Comment (klausi) [inline src/Plugin/RulesAction/SystemMessage.php]: should be "bool"

**Comment by ndeet** [comment]
done

File: src/Plugin/RulesAction/SystemMessage.php [hunk 4]
```diff
@@ -37,14 +37,20 @@
...
   /**
-   * {@inheritdoc}
+   * Set a system message.
+   *
+   * @param string $message
+   *   Message string that should be set.
+   * @param string $type
+   *   Type of the message.
+   * @param bool $repeat
+   *   Boolean if the message should be repeated.
```
Comment (klausi) [inline src/Plugin/RulesAction/SystemMessage.php]: should be prefixed with "(Optional)". Comment should be "TRUE if the message should be repeated."

**Comment by ndeet** [comment]
Fixed, i used "(optional)" - with lowercase "o" - as all other occurrences in the module seem to use lowercase.

**Comment by klausi** [comment]
merged, thanks!

**Pull request closed without merging**